Doc/whatsnew/3.13.rst

```diff
@@ -265,223 +264,233 @@ Improved Error Messages
-(including debuggers and code execution tracing tools) potentially resetting the shared
-snapshot in that scope. Now, the code will always run against an independent snapshot of the
-local variables in optimized scopes, and hence the changes will never be visible in
-subsequent calls to ``locals()``. To access the changes made in these cases, an explicit
-namespace reference must now be passed to the relevant function. Alternatively, it may make
-sense to update affected code to use a higher level code execution API that returns the
-resulting code execution namespace (e.g. :func:`runpy.run_path` when executing Python
-files from disk).
+.. _whatsnew313-free-threaded-cpython:
 
-To ensure debuggers and similar tools can reliably update local variables in
-scopes affected by this change, :attr:`FrameType.f_locals <frame.f_locals>` now
-returns a write-through proxy to the frame's local and locally referenced
-nonlocal variables in these scopes, rather than returning an inconsistently
-updated shared  ``dict`` instance with undefined runtime semantics.
+Free-threaded CPython
+---------------------
 
-See :pep:`667` for more details, including related C API changes and deprecations. Porting
-notes are also provided below for the affected :ref:`Python APIs <pep667-porting-notes-py>`
-and :ref:`C APIs <pep667-porting-notes-c>`.
+CPython now has experimental support for running in a free-threaded mode,
+with the :term:`global interpreter lock` (GIL) disabled.
+This is an experimental feature and therefore is not enabled by default.
+The free-threaded mode requires a different executable,
+usually called ``python3.13t`` or ``python3.13t.exe``.
+Pre-built binaries marked as *free-threaded* can be installed as part of
+the official :ref:`Windows <install-freethreaded-windows>`
+and :ref:`macOS <getting-and-installing-macpython>` installers,
+or CPython can be built from source with the :option:`--disable-gil` option.
 
-(PEP and implementation contributed by Mark Shannon and Tian Gao in
-:gh:`74929`. Documentation updates provided by Guido van Rossum and
-Alyssa Coghlan.)
-
-Incremental Garbage Collection
-------------------------------
+.. better macOS link pending
+   https://github.com/python/cpython/issues/109975#issuecomment-2286391179
 
-* The cycle garbage collector is now incremental.
-  This means that maximum pause times are reduced
-  by an order of magnitude or more for larger heaps.
+Free-threaded execution allows for full utilization of the available
+processing power by running threads in parallel on available CPU cores.
+While not all software will benefit from this automatically, programs
+designed with threading in mind will run faster on multi-core hardware.
+**The free-threaded mode is experimental** and work is ongoing to improve it:
+expect some bugs and a substantial single-threaded performance hit.
+Free-threaded builds of CPython support optionally running with the :term:`GIL`
```

Comment:
We only need to link the first occurrence in this section

https://devguide.python.org/documentation/style-guide/#links

```suggestion
Free-threaded builds of CPython support optionally running with the GIL
```